lib/scoreboard.rb

```diff
@@ -0,0 +1,42 @@
+class Scoreboard
+  def initialize(rolls)
+    @rolls = rolls
+  end
+
+  def score
+    index = 0
+    score = 0
+    10.times do
+      if strike?(index)
+        score += 10 + strike_bonus_points(index)
+      elsif spare?(index)
+        score += 10 + spare_bonus_points(index)
+      else
+        score += regular_points(index)
+      end
+      strike?(index) ? index += 1 : index += 2
+    end
+    score
```

Comment:
Honestly your code is so DRY and a pleasure to read that I really am being picky here to try and find something to comment on! 
Perhaps using a different variable name to the name of the method (score) might keep this code a bit more readable and avoid accidentally calling the score method instead of returning the score variable here - just thinking in some languages this might recursively call the score method?